### PR TITLE
mon: remove ceph-create-keys from mon startup

### PR DESCRIPTION
**Pull request by oms4suse**
Fixes: http://tracker.ceph.com/issues/16036

### CODE REVIEW

**Comment by oms4suse** [comment]
I suspect the issue is in the test suite as the failing command is:

    ceph osd setcrushmap -i /tmp/cephtool26589/map

Since now the mon servers will not definitely have admin keyrings on them this could be he reason for a false positive.

**Comment by liewegas** [comment]
test this please

**Comment by oms4suse** [comment]
Seems all test have past this time :)

**Comment by yuriw** [comment]
@oms4suse failed to build http://gitbuilder.sepia.ceph.com/gitbuilder-ceph-deb-trusty-amd64-notcmalloc/log.cgi?log=1f26a813a0ceef144afa25a86a029e67ad70c665

install: cannot stat ‘systemd/ceph-create-keys@.service’: No such file or directory

**Comment by smithfarm** [comment]
@oms4suse Needs rebase

**Comment by smithfarm** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;wQs2xZWU&#x2F;1915-drop-ceph-create-keys-service"> Drop ceph-create-keys@.service</a>

**Comment by liewegas** [comment]
@oms4suse ping

**Comment by oms4suse** [comment]
Hi sage, I thought this was resolved upstream already

**Comment by oms4suse** [comment]
I am back on this PR.

**Comment by oms4suse** [comment]
Pushed new version.

**Comment by oms4suse** [comment]
Rebased to master again

**Comment by oms4suse** [comment]
Rebasing to restart checks.

Jenkins agent failed last time, and http request failed, both unrelated to this code base.

**Comment by tchaikov** [comment]
jenkins, retest this please.

**Comment by oms4suse** [comment]
seems all tests passed.

**Comment by oms4suse** [comment]
@yuriw Please can you retest?

**Comment by smithfarm** [comment]
Needs rebase again.

**Comment by oms4suse** [comment]
Thanks Nathan I will get back to this.

**Comment by oms4suse** [comment]
Rebased to master

**Comment by liewegas** [comment]
@yuriw let's get this merged this week for kraken.  probably just needs a ceph-deploy and smoke test?

**Comment by oms4suse** [comment]
Test failed but I think its unrelated.

**Comment by oms4suse** [comment]
Rebased on todays master to see if tests now pass

**Comment by oms4suse** [comment]
Seems tests pass on todays master.

**Comment by oms4suse** [comment]
@yuriw Please can you retest?

**Comment by liewegas** [comment]
This passes my manual ceph-deploy test.

File: ceph.spec.in [hunk 1]
```diff
@@ -809,7 +809,7 @@ rm -rf %{buildroot}
 %{_bindir}/osdmaptool
 %{_bindir}/ceph-run
 %{_bindir}/ceph-detect-init
-%{_unitdir}/ceph-create-keys@.service
+%{_bindir}/ceph-client-debug
```
Comment (liewegas) [inline ceph.spec.in]: I think this line shouldn't be added here... can you remove?

Comment (oms4suse) [inline ceph.spec.in]: Sure

**Comment by oms4suse** [comment]
Changes requested complete I think.

**Pull request merged**